src/storage/mod.rs

```diff
@@ -812,15 +853,37 @@ impl Storage {
         &self,
         ctx: Context,
         key: Vec<u8>,
-        callback: Callback<Option<Vec<u8>>>,
-    ) -> Result<()> {
-        let cmd = Command::RawGet {
-            ctx: ctx,
-            key: Key::from_encoded(key),
-        };
-        self.schedule(cmd, StorageCb::SingleValue(callback))?;
-        RAWKV_COMMAND_COUNTER_VEC.with_label_values(&["get"]).inc();
-        Ok(())
+    ) -> Box<Future<Item = Option<Vec<u8>>, Error = Error> + Send> {
+        static CMD: &'static str = "raw_get";
+        let engine = self.get_engine();
+        let priority = readpool::Priority::from(ctx.get_priority());
+
+        let res = self.read_pool.future_execute(priority, move |ctxd| {
+            let _timer = {
+                let ctxd = ctxd.clone();
+                let mut thread_ctx = ctxd.current_thread_context_mut();
+                thread_ctx.collect_command_count_and_duration(CMD, priority, true)
+            };
+
+            Self::async_snapshot(engine, &ctx).and_then(move |snapshot: Box<Snapshot>| {
+                let mut thread_ctx = ctxd.current_thread_context_mut();
+                let _t_process = thread_ctx.collect_processing_read_duration(CMD);
+
+                // no scan_count for this kind of op.
+
+                snapshot.get(&Key::from_encoded(key))
+                        // map storage::engine::Error -> storage::Error
+                        .map_err(Error::from)
+                        .map(|r| {
+                            thread_ctx.collect_key_reads(CMD, 1);
```

Comment:
How about read bytes?

Reply:
Ok. It is not collected previously so that it is not collected here as well.

Reply:
Previously only MvccReader collects read bytes. We can set up another PR dealing with this issue.